src/main/java/ua/softserveinc/tc/dao/BaseDao.java

```diff
@@ -1,17 +1,14 @@
 package ua.softserveinc.tc.dao;
 
-import javax.persistence.EntityManager;
 import java.util.List;
 
 public interface BaseDao<T> {
-    EntityManager getEntityManager();
```

Comment:
I think it brakes IoC principle becase it exposes ```EntityManager```, which is JPA internal interface, so we coupled with it. And if we decide to change the impelementation, this method will make no sense.